server/geoip_service.py

```diff
@@ -193,11 +193,16 @@ def load_db(self) -> None:
             # Set the time first, if the file is corrupted we don't need to try
             # loading it again anyways
             self.db_update_time = datetime.now()
-            self.db = geoip2.database.Reader(self.file_path)
+            new_db = maxminddb.open_database(self.file_path)
+
+            if self.db is not None:
+                self.db.close()
```

Comment:
Is it worth awaiting your `shutdown` function here and making this function `async`?

Reply:
Nah, `shutdown` is a lifecycle hook for services that is called when the server is being stopped. It would be bad style to call it from elsewhere. I mean I could make a `self.close` method and call if from shutdown and here, but that seems a bit overkill for refactoring 2 lines :P.